backend/api/utils.py

```diff
@@ -0,0 +1,105 @@
+from project import models as project_models
+from user import models as user_models
+from django.db.models import Avg, Window
+
+def user_json(user, user_id=""):
+    obj = {}
+    obj['id'] = str(user.id)
+    obj['full_name'] = user.full_name
+    obj['username'] = user.username
+    obj['email'] = user.email
+    obj['type'] = user.type
+    obj['gender'] = user.gender
+    obj['bio'] = user.bio
+    obj['profile_image'] = user.profile_image
+    obj['created_at'] = user.created_at
+    obj['updated_at'] = user.updated_at
+    ratings_rater = user_models.Rating.objects.filter(rater=user)
+    ratings_rated = user_models.Rating.objects.filter(rated=user)
+    obj['ratings'] = {}
+    obj['ratings']['rater'] = []
+    for rating in ratings_rater:
+        obj['ratings']['rater'].append(rating_json(rating, user_id, "user"))
+    obj['ratings']['rated'] = []
+    for rating in ratings_rated:
+        obj['ratings']['rated'].append(rating_json(rating, user_id, "user"))
+    avg1 = ratings_rater.average('value')
+    avg2 = ratings_rated.average('value')
+    obj['avg_rating'] = (avg1 + avg2) / 2
```

Comment:
why do you combine 2 rating types: rater and rated?

Reply:
I was aiming to give only one average rating for a user. But we also can give multiple averages, if you prefer.

Reply:
I think we should show only rated average.